docs/json.md

```diff
@@ -0,0 +1,421 @@
+
+We introduced API 4.0 to guarantee the JSON payload response types match the API 4.0 specification.
+
+## Lookering forward
+
+Current users of Looker SDKs may have noticed an `id` type may be either `integer` or `string`. For the stable (aka GA, generally available) release of API 4.0, all `id` references will be typed as `string`.
+Using strings for all ID references will allow Looker to scale out Looker services in the future, once ID generation does not require an auto-increment numeric ID value.
+Also, API users don't need to be concerned with the fact that an ID is a numeric value internally.
+
+Theoretically, changing the type of entity IDs could be extremely disruptive when API 4.0 is released.
+
+We have worked on minimizing this disruption for all Looker-provided language SDKs. This document describes some JSON parsing requirements and how our SDKs support them.
+
+## JSON parsing requirements
+
+There are many requirements for parsing and producing JSON, but most of the issues are identical among languages that support JSON **serialization** and **deserialization**.
+Other names for this process are **marshalling** and **unmarshalling** or **encoding** and **decoding**, but they all mean basically the same thing.
+
+Serialization is the conversion of a JSON payload into a structure, record, or object of a given language. Deserialization is rendering an instantiated data type to its JSON representation.
+
+Almost all languages include JSON serialization and deserialization support. Many include more than one, with an offering from a third-party as the de facto standard, rather than the first-party solution provided by the language vendor.
+
+The parsing requirements of concern for the Looker language SDKs are:
+
+1. JSON value types conflicting with the API type specification. This is called **type ambiguity** below. For the Looker API, these conflicts should be automatically resolvable for our SDKs. The conflict is either:
+   1. a numeric value for a string property (forward compatibility issue)
+   2. a string value for an integer property (backward compatibility issue)
+2. New properties in a JSON payload. These are ignored by the SDKs. (See [TypeScript caveats](#typescript-caveats) for additional comments.)
+3. Removed properties in a JSON payload. Typically, the property will end up unassigned. Some SDKs may throw errors with missing required properties, but that client code should be changed anyway if the property no longer exists.
+4. Name changes, like `space` to `folder` and `homepage` to `board`. There is no reasonable automatic solution for this, so type or property renames will require changing SDK existing code.
```

Comment:
```suggestion
Looker SDK JSON deserializing requirements are: 

1. Resolve **type ambiguity** where JSON value types conflict with the API type specification. We further discuss **type ambiguity** later in this document. The specific type conflicts are:
   1. Deserializing a numeric JSON type to a `string` type property (forward compatibility issue)
   2. Deserializing a string JSON type to an `integer` type property (backward compatibility issue)
2. Ignore unknown properties in a JSON payload. (See [TypeScript caveats](#typescript-caveats) for additional comments.)
3. Leave missing properties in a JSON payload unassigned. Some SDKs may throw errors with missing required properties, but you should update your client code if the required property no longer exists.
4. Handle property and type name changes, like `space` to `folder` and `homepage` to `board`. There is no reasonable automatic solution for this, so will require updating your existing SDK.
```

Reply:
Item 4 doesn't meet the changed criteria you made on line 32. It's not a requirement of our SDKs.